crates/nu-cli/src/commands/first.rs

```diff
@@ -57,27 +57,27 @@ impl WholeStreamCommand for First {
     }
 }
 
-fn first(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
+async fn first(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
     let registry = registry.clone();
-    let stream = async_stream! {
-        let (FirstArgs { rows }, mut input) = args.process(&registry).await?;
-        let mut rows_desired = if let Some(quantity) = rows {
-            *quantity
-        } else {
-            1
-        };
+    let (FirstArgs { rows }, mut input) = args.process(&registry).await?;
+    let mut rows_desired = if let Some(quantity) = rows {
+        *quantity
+    } else {
+        1
+    };
+
+    let mut values_vec_deque = VecDeque::new();
 
-        while let Some(input) = input.next().await {
-            if rows_desired > 0 {
-                yield ReturnSuccess::value(input);
-                rows_desired -= 1;
-            } else {
-                break;
-            }
+    while let Some(input) = input.next().await {
+        if rows_desired > 0 {
+            values_vec_deque.push_back(ReturnSuccess::value(input));
+            rows_desired -= 1;
+        } else {
+            break;
```

Comment:
You could even get this a bit tighter and keep the streaming capability, I think. Instead of calling `input.next().await`, I think you could do something like:

`Ok(input.first(rows_desired).to_output_stream())`